### PR TITLE
fix (directchat): auto disconnect self once conn closed

### PR DESCRIPTION
**Pull request by emizzle**
When closing a connection of clientA(using the command “/disconnect”), the connection is then closed on clientA’s side. However, the connection remained open on clientB and clientB could continue sending message to clientA and clientA would receive those messages.

This PR listens for a closing connection on clientB, then also closes clientB’s connection.

### CODE REVIEW

**Review by Menduist — CHANGES_REQUESTED** [review]
https://github.com/status-im/nim-libp2p/pull/662 needs to be merged first, blocking here in the meantime

**Comment by arnetheduck** [comment]
this looks off  - generally, closing is a two-phase operation where one side initiates the close and the other side performs on receiving the close message and does cleanup (which might include sending more messages) then closes its side - automatic closing prevents that from happening.

**Comment by emizzle** [comment]
> https://github.com/status-im/nim-libp2p/pull/662 needs to be merged first, blocking here in the meantime 

Because #662 was closed, this has been rebased on top of unstable.

**Review by dryajov — APPROVED** [review]
LGTM. This can be merged regardless of #662.

File: examples/directchat.nim [hunk 1]
```diff
@@ -79,7 +79,9 @@ proc handlePeer(c: Chat, conn: Connection) {.async.} =
       c.writeStdout $conn.peerId & ": " & $str
 
   except LPStreamEOFError:
-    c.writeStdout $conn.peerId & " disconnected"
+    defer: c.writeStdout $conn.peerId & " disconnected"
```
Comment (dryajov) [inline examples/directchat.nim:82]: Just a word of caution with `defer`, which is known to fail when combined with `async`.

**Review by Menduist — APPROVED** [review]

**Pull request merged**